ios/InstagramShare.m

```diff
@@ -50,4 +51,74 @@ - (void)shareSingle:(NSDictionary *)options
     } 
 }
 
+- (void)shareSingleImage:(NSDictionary *)options
+         failureCallback:(RCTResponseErrorBlock)failureCallback
+         successCallback:(RCTResponseSenderBlock)successCallback {
+    
+    UIImage *image;
+    NSURL *imageURL = [RCTConvert NSURL:options[@"url"]];
+    if (imageURL) {
+        if (imageURL.fileURL || [imageURL.scheme.lowercaseString isEqualToString:@"data"]) {
+            NSError *error;
+            NSData *data = [NSData dataWithContentsOfURL:imageURL
+                                                 options:(NSDataReadingOptions)0
+                                                   error:&error];
+            if (!data) {
+                failureCallback(error);
+                return;
+            }
+            image = [UIImage imageWithData: data];
+            [self savePictureAndOpenInstagram: image
+                              failureCallback: failureCallback
+                              successCallback: successCallback];
+        }
+    } else {
+        [[UIApplication sharedApplication] openURL: [NSURL URLWithString:@"instagram://camera"]];
+        successCallback(@[]);
+    }
+}
+
+-(void)savePictureAndOpenInstagram:(UIImage *)base64Image
+                   failureCallback:(RCTResponseErrorBlock)failureCallback
+                   successCallback:(RCTResponseSenderBlock)successCallback {
+    
+    NSURL *URL = [self fileURLWithTemporaryImageData:UIImageJPEGRepresentation(base64Image, 0.9)];
```

Comment:
Is this recompressing the image before sending it to Instagram?

Reply:
it seems so